internal/venice-client-common/src/main/java/com/linkedin/venice/stats/metrics/MetricEntityStateGeneric.java

```diff
@@ -0,0 +1,127 @@
+package com.linkedin.venice.stats.metrics;
+
+import static com.linkedin.venice.stats.VeniceOpenTelemetryMetricsRepository.REDUNDANT_LOG_FILTER;
+
+import com.linkedin.venice.stats.VeniceOpenTelemetryMetricsRepository;
+import com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions;
+import io.opentelemetry.api.common.Attributes;
+import io.tehuti.metrics.MeasurableStat;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * Provides a flexible, generic, and non-caching implementation of {@link MetricEntityState} for zero or more
+ * dynamic dimensions, where the dimensions can be enums or arbitrary strings
+ *
+ * This implementation should be used only in certain control-path components (such as controllers) where:
+ * 1. Performance constraints are lenient, and metric recording is infrequent.
+ * 2. The metric entity requires an arbitrary number of dynamic dimensions without predefined enums.
+ * This approach also helps reduce code complexity and avoids the proliferation of specialized subclasses (like
+ * MetricEntityStateOneEnum or MetricEntityStateTwoEnums) for different dynamic dimension combinations when
+ * attribute caching is not necessary.
+ *
+ * Compared to enum-based subclasses, this class provides less compile-time type safety, as it does not enforce
+ * dynamic dimensions to be enums and does not require explicit dimension types during instantiation.
+ */
+public class MetricEntityStateGeneric extends MetricEntityState {
+  public MetricEntityStateGeneric(
+      MetricEntity metricEntity,
+      VeniceOpenTelemetryMetricsRepository otelRepository,
+      Map<VeniceMetricsDimensions, String> baseDimensionsMap) {
+    this(metricEntity, otelRepository, null, null, Collections.EMPTY_LIST, baseDimensionsMap);
+  }
+
+  public MetricEntityStateGeneric(
```

Comment:
Seems like we use a factory method in `MetricEntityStateOneEnum` for object creation. Do we want to be consistent in this class?

Reply:
Makes sense. Fixed here as well as in `MetricEntityStateBase`.